### PR TITLE
Add C++ client template

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR:
- Adds an empty implementation of the C++ Oak Client
- Adds C++ gRPC to the WORKSPACE

### CODE REVIEW

**Review by tiziano88 — APPROVED** [review]

File: cc/transport/BUILD [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    default_visibility = ["//visibility:public"],
+    licenses = ["notice"],
+)
+
+cc_library(
+    name = "transport",
+    srcs = ["transport.cc"],
+    hdrs = ["transport.h"],
+    deps = [
+        "//oak_remote_attestation/proto/v1:messages_java_proto",
```
Comment (pmcgrath17) [inline cc/transport/BUILD:27]: messages_cc_proto?

File: cc/transport/BUILD [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+)
+
+cc_library(
+    name = "grpc_streaming_transport",
+    srcs = ["grpc_streaming_transport.cc"],
+    hdrs = ["grpc_streaming_transport.h"],
+    deps = [
+        "//oak_remote_attestation/proto/v1:messages_java_proto",
+        "//oak_remote_attestation/proto/v1:service_streaming_java_proto",
+    ],
```
Comment (pmcgrath17) [inline cc/transport/BUILD:38]: use the cc_proto dependencies here

**Review by pmcgrath17 — APPROVED** [review]

File: cc/transport/BUILD (hunk 1 above)
Comment (ipetr0v) [reply cc/transport/BUILD:27]: Done

File: cc/transport/BUILD (hunk 2 above)
Comment (ipetr0v) [reply cc/transport/BUILD:38]: Done

**Comment by ipetr0v** [comment]
Currently blocked by https://github.com/project-oak/oak/pull/4056

**Pull request merged**